### DISCUSSION TITLE
Communication in the web3Modal backend is annoying

### DISCUSSION THREAD

**Discussion by dangdang02** [WalletConnect/walletconnect-monorepo#4015] [Developer Support]
POST https://mainnet.infura.io/v3/84842078b09946638c03157f83405213 net::ERR_CONNECTION_TIMED_OUT

Uncaught (in promise) Error: Socket stalled when trying to connect to wss://relay.walletconnect.org
{context: 'core'}
{context: 'core/relayer'}
 Error: Socket stalled when trying to connect to wss://relay.walletconnect.com
    at jsonrpc.ts:29:23

----------------------------------------------

I have not enabled the mainnet, and I am solely using Arbitrum. Why connect to https://mainnet.infura.io/v3/84842078b09946638c03157f83405213?

I have not enabled enableAnalytics, so why connect to wss://relay.walletconnect.com?

I don't want users to connect and communicate with any URL other than the rpcUrl when using web3modal; this would make users feel insecure.

Is it possible to disable connections and communication with all URLs outside of the custom rpcUrl?

**Comment by glitch-txs** [comment]
They relay is used for the WalletConnect protocol, we have a current alpha where you can disable it but all WalletConnect supported wallets won't work with your dapp, you also need to config a project ID for the relay

**Comment by dangdang02** [comment]
Thank you for your reply. I have successfully connected with WalletConnect, but I am having difficulty understanding the connection to https://mainnet.infura.io/v3/84842078b09946638c03157f83405213. This is neither the rpcUrl I have set nor does it seem to be related to WalletConnect functionality.

**Reply by glitch-txs** [reply]
can you tell me the library you're using and version?

**Comment by dangdang02** [comment]
web3modal 3, the latest version

"@web3modal/ethers5": "^3.4.0",
"ethers": "^5.7.2",

**Reply by glitch-txs** [reply]
I see, it's a public endpoint used to fetch the ENS name, if you'd like the option to disable it when using other chains please open an issue in the repo https://github.com/WalletConnect/web3modal/issues

**Comment by dangdang02** [comment]
Thanks
